src/js/grid.js

```diff
@@ -62,20 +62,22 @@
     /**
      * Initializes the grid.
      */
-    init() {
-      this._node = document.getElementById("newtab-grid");
-      this._createSiteFragment();
-      addEventListener("resize", this);
+    init: async(function*() {
+      gGrid._node = document.getElementById("newtab-grid");
+      gGrid._createSiteFragment();
 
-      gNewTab.sendToBrowser("NewTab:InitializeGrid");
-      gNewTab.registerListener("NewTab:InitializeLinks",
-        this.refresh.bind(this));
+      gGrid._sw = (yield navigator.serviceWorker.ready).active;
```

Comment:
We might not need to store the sw in a private variable, as it's only ever used here.